pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
"NoInput" seems even vaguer/more confusing than "isScratch".  I had to think for a few minutes about why you were suggesting "NoInput" before I understood that a build based on a scratch image has "no input image" (as opposed to one based on FROM centos, or one based on an imagestream).

And definitely not all "LocalOnly" refs would be "NoInput" refs, as evidenced by this exact scenario (the image is only available locally, but the build does require an input, hence we must warn the user their input may not be available)

(If NoInput is intended to refer to an existing flag, I couldn't find it)